src/backend_common.cc

```diff
@@ -1019,20 +1019,26 @@ BatchInput::ParseFromModelConfig(
       }
     }
     {
-      std::string bi_kind;
-      RETURN_IF_ERROR(bi.MemberAsString("kind", &bi_kind));
-      if (bi_kind == "BATCH_ELEMENT_COUNT") {
+      RETURN_IF_ERROR(bi.MemberAsString("kind", &batch_input.kind_str_));
```

Comment:
How about creating a new function in BatchInput class that takes in bi and populates the required fields. This would avoid the struct like dereferencing used here.